### DISCUSSION TITLE
Global forms permission for new forms

### DISCUSSION THREAD

**Discussion by kylewalow** [statamic/cms#4755] [Q&A] [answered]
We have a dynamic formbuilder where the user can create his own forms.

At the moment every new form creates a new line in the permissions. If a no super-admin creates a new from he has to ask a super-admin to give him the permission to edit the form.

Is there a possibility to give forms permission on default for each form even for new onces?

**Comment by duncanmcclean** [comment, marked as answer]
This can be done with a little bit of extra code, here's what I did earlier:

1. Create a file (you can change it later): `app/Listeners/AddFormPermissionsToAdminRole.php`

```php
<?php

namespace App\Listeners;

use Illuminate\Contracts\Queue\ShouldQueue;
use Illuminate\Queue\InteractsWithQueue;
use Statamic\Events\FormSaved;
use Statamic\Facades\Role;

class AddFormPermissionsToAdminRole
{
    /**
     * Create the event listener.
     *
     * @return void
     */
    public function __construct()
    {
        //
    }

    /**
     * Handle the event.
     *
     * @param  object  $event
     * @return void
     */
    public function handle(FormSaved $event)
    {
        $adminRole = Role::find('site_admin');

        if (! $adminRole->hasPermission("view {$event->form->handle()} form submissions")) {
            $adminRole
                ->addPermission("view {$event->form->handle()} form submissions")
                ->addPermission("delete {$event->form->handle()} form submissions")
                ->save();
        }
    }
}
```

Make sure to replace `site_admin` with the handle of a user role.

2. In your `app/Providers/EventServiceProvider.php` file, add the below to the `$listen` array:

```php
\Statamic\Events\FormSaved::class => [
    \App\Listeners\AddFormPermissionsToAdminRole::class,
],
```

Then when any new forms are created, your user should have access to edit them.

**Reply by kylewalow** [reply]
Than you it works like a charm

**Reply by duncanmcclean** [reply]
Fantastic!
